test/TensorFlowRuntime/tensor_autodiff_runtime.swift

```diff
@@ -117,6 +117,27 @@ TensorADTests.testAllBackends("mean") {
   expectEqual(expected, meanGradAlongAxes(input))
 }
 
+TensorADTests.testAllBackends("expandingShape") {
+  let f1 = { (a: Tensor<Float>) in a.expandingShape(at: 0).squared() }
+  let f2 = { (a: Tensor<Float>) in a.squared().expandingShape(at: 0) }
+  expectEqual([6, 10], gradient(at: [3, 5], in: f1))
```

Comment:
`gradient` should be used only for functions that return scalar tensors (i.e. tensors with rank 0).
However, `f1` and `f2` do not return scalar tensors.

Please modify tests to call `pullback` instead, and pass in a direction vector of ones with the same shape as the output:
```suggestion
  expectEqual([6, 10], pullback(at: [3, 5], in: f1)([[1, 1]]))
  expectEqual([6, 10], pullback(at: [3, 5], in: f2)([[1, 1]]))
```